win32ss/gdi/ntgdi/d3dkmt.c

```diff
@@ -0,0 +1,706 @@
+#include <win32k.h>
+#include <reactos/rddm/rxgkinterface.h>
+#include <debug.h>
+
+/*
+ * It looks like windows saves all the funciton pointers globally inside win32k,
+ * Instead, we're going to keep it static to this file and keep it organized in struct
+ * we obtained with the IOCTRL.
+ */
+static REACTOS_WIN32K_DXGKRNL_INTERFACE DxgAdapterCallbacks = {0};
+
+/*
+ * This looks like it's done inside DxDdStartupDxGraphics, but i'd rather keep this organized.
+ * Dxg gets start inveitibly anyway it seems atleast on vista.
+ */
+VOID
+DxStartupDxgkInt()
+{
+    DPRINT("DxStartupDxgkInt: Entry\n");
+    //TODO: Let DxgKrnl know it's time to start all adapters, and obtain the win32k<->dxgkrnl interface via an IOCTRL.
+}
+
+BOOLEAN
+APIENTRY
+NtGdiDdDDICheckExclusiveOwnership(VOID)
+{
+    // We don't support DWM at this time, excusive ownership is always false.
+    return FALSE;
+}
+
+NTSTATUS
+APIENTRY
+NtGdiDdDDIGetProcessSchedulingPriorityClass(_In_  HANDLE                                unnamedParam1,
+                                            _Out_ D3DKMT_SCHEDULINGPRIORITYCLASS        *unnamedParam2)
+{
+    return 1;
+}
+
+NTSTATUS
+APIENTRY
+NtGdiDdDDISetProcessSchedulingPriorityClass(
+                    _In_ HANDLE                                     unnamedParam1,
+                    _In_ D3DKMT_SCHEDULINGPRIORITYCLASS             unnamedParam2)
+{
+    return 1;
+}
+
+NTSTATUS
+APIENTRY
+NtGdiDdDDISharedPrimaryLockNotification(_In_ const D3DKMT_SHAREDPRIMARYLOCKNOTIFICATION* unnamedParam1)
+{
+    return 1;
+}
+
+NTSTATUS
+APIENTRY
+NtGdiDdDDISharedPrimaryUnLockNotification(_In_ const D3DKMT_SHAREDPRIMARYUNLOCKNOTIFICATION* unnamedParam1)
+{
+    return 1;
+}
+
+NTSTATUS
+APIENTRY
+NtGdiDdDDIOpenAdapterFromGdiDisplayName(_Inout_ D3DKMT_OPENADAPTERFROMGDIDISPLAYNAME* unnamedParam1)
+{
+   return 0;
+}
+
+NTSTATUS
+APIENTRY
+NtGdiDdDDIOpenAdapterFromHdc(_Inout_ D3DKMT_OPENADAPTERFROMHDC* unnamedParam1)
+{
+    return 0;
+}
+
+
+NTSTATUS
+APIENTRY
+NtGdiDdDDIOpenAdapterFromDeviceName(_Inout_ D3DKMT_OPENADAPTERFROMDEVICENAME* unnamedParam1)
+{
+    return 0;
+}
+
+
+/*
+ * The following APIs all have the same idea,
+ * Most of the parameters are stuffed in custom typedefs with a bunch of types inside them.
+ * The idea here is this:
+ * if we're dealing with a d3dkmt API that directly calls into a miniport if the function pointer doesn't
+ * exist we're returning STATUS_PROCEDURE_NOT_FOUND.
+ *
+ * This essentially means the Dxgkrnl interface was never made as Win32k doesn't do any handling for these routines.
+ */
+
+NTSTATUS
+APIENTRY
+NtGdiDdDDICreateAllocation(_Inout_ D3DKMT_CREATEALLOCATION* unnamedParam1)
```

Comment:
`unnamedParam1` ? Where is that from?

Reply:
https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/d3dkmthk/nf-d3dkmthk-d3dkmtcreateallocation

The usermode exposed version of these functions refer to it as unnamedParam1, 2 etc.
So I got it from MSDN